src/components/modals/EntityCreatorEditor/EntityCreatorContainer.tsx

```diff
@@ -359,8 +355,10 @@ class Container extends React.Component<Props, ComponentState> {
             return Util.formatMessageId(intl, FM.ENTITYCREATOREDITOR_FIELDERROR_ALPHANUMERIC)
         }
 
-        if (Object.values(this.state.restricted_entity_names).indexOf(value.toLocaleLowerCase()) >= 0) {
-            return Util.formatMessageId(intl, FM.ENTITYCREATOREDITOR_FIELDERROR_RESERVED)
+        if (!isNameUnrestricted) {
+            if ("builtin-" === value.toLocaleLowerCase().substring(0,"builtin-".length))  {
```

Comment:
Save the "builtin-" string as a variable and use it here and the `GetPrebuiltEntityName`.  The guarantees the prefix can't be out of sync.